src/Amadeus/Client/Struct/Security/PasswordInfo.php

```diff
@@ -43,9 +44,22 @@ class PasswordInfo
      *
      * See Amadeus Core Webservices documentation
      * [DATA TYPE codesets (Ref: 116Z 1A 02.1.8)]
+     *
      * @var string
      */
     const DATA_TYPE_BINARY = 'B';
+    /**
+     * When using "EDIFACT DATA" type, password length value must be 12.
+     *
+     * @var int
+     */
+    const PASSWORD_LENGTH_EDIFACT = 12;
+    /**
+     * When using "BINARY DATA" type, password length value must be 40.
```

Comment:
Are you sure about this? AFAIK this is just the length of the plaintext password, but I don't know if Amadeus uses fixed length passwords (which would not be very smart)

Reply:
For me it was a big surprise.

I used `strlen` and request was:
```
<ns1:passwordInfo>
    <ns1:dataLength>16</ns1:dataLength>
    <ns1:dataType>E</ns1:dataType>
    <ns1:binaryData>UFlOU2ZYV3F6MKdR</ns1:binaryData>
</ns1:passwordInfo>
```
But response was:
```
<freeText>Bad Grammar:  &lt;From: InboundEdifactMessageImpl.cpp#248&gt; Exception caug</freeText>
<freeText>ht while decoding Fusion DSC found an exception !  Binary BLB segment </freeText>
<freeText>  The following structure is expected for BLB:   BLB+binaryLength+data</freeText>
<freeText>Type+binaryData&apos; or its equivalent in xml   with the following restric</freeText>
<freeText>tions :   binaryLength : numeric, must be the exact size in bytes of b</freeText>
<freeText>inaryData   dataType: optional, length 1   binaryData: not checked  Cu</freeText>
<freeText>rrent position in buffer: +PYNSfXWqz5GQ&apos;  Last 15 characters before er</freeText>
<freeText>ror: DUT:SU&apos;BLB+16+E  Current path in the grammar : /passwordInfo#1</freeText>
```

I checked everything several times and also tried with SHA1 hashing of password (which length was bigger than 40).

Amadeus Support replied to me that data length value should be 12.
After I set this fixed data length - authenticate is successful.

Reply:
Whaaaa, okay, thanks for that

Reply:
Yeah, it's stange

![image](https://user-images.githubusercontent.com/21292595/47458734-ffdc2b80-d7e2-11e8-83f8-9c2d8500b661.png)

Reply:
For type E the password length should be the length of the password before it is base64 encoded